### PR TITLE
[hiredis] Support static build on Windows

### PR DESCRIPTION
**Pull request by NancyLi1013**
Add windows support. 
It currently only supports static on Windows.

Related issue #8840 
Related PR #8843, #8846.
Note: All features test pass with x64-linux triplet.
All features are only supported on Unix.

### CODE REVIEW

File: ports/hiredis/portfile.cmake [hunk 1]
```diff
@@ -1,21 +1,34 @@
-vcpkg_fail_port_install(ON_TARGET "Windows")
+if (VCPKG_TARGET_IS_WINDOWS AND VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
```
Comment (JackBoosY) [inline ports/hiredis/portfile.cmake]: We can use:
```
if (VCPKG_TARGET_IS_WINDOWS)
    vcpkg_fail_port_install(ON_LIBRARY_LINKAGE  "dynamic")
endif()
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

File: ports/hiredis/portfile.cmake (hunk 1 above)
Comment (NancyLi1013) [reply ports/hiredis/portfile.cmake]: I updated as the following code:

```
if(VCPKG_TARGET_IS_WINDOWS)
    vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
endif()
```

**Comment by NancyLi1013** [comment]
The Related PR #9388 has been merged.
So I close this PR now.

**Pull request closed without merging**